heft-plugins/heft-lint-plugin/src/Eslint.ts

```diff
@@ -141,57 +238,74 @@ export class Eslint extends LinterBase<TEslint.ESLint.LintResult> {
     }
   }
 
-  protected async getCacheVersionAsync(): Promise<string> {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const eslintBaseConfiguration: any = await this._linter.calculateConfigForFile(
-      this._linterConfigFilePath
+  protected override async getCacheVersionAsync(): Promise<string> {
+    return `${this._eslintPackageVersion.version}_${process.version}`;
+  }
+
+  protected override async getSourceFileHashAsync(sourceFile: IExtendedSourceFile): Promise<string> {
+    const sourceFileEslintConfiguration: IExtendedEslintConfig = await this._linter.calculateConfigForFile(
+      sourceFile.fileName
     );
-    const eslintConfigHash: crypto.Hash = crypto
-      .createHash('sha1')
-      .update(JSON.stringify(eslintBaseConfiguration));
-    const eslintConfigVersion: string = `${this._eslintPackage.Linter.version}_${eslintConfigHash.digest(
-      'hex'
-    )}`;
-
-    return eslintConfigVersion;
+
+    // The eslint configuration object contains a toJSON() method that returns a serializable version of the
+    // configuration. However, we are manually injecting the TypeScript program into the parserOptions, which
+    // is not serializable. Patch the function to remove the program before returning the serializable version.
```

Comment:
What happens if we instead inject it using `Object.defineProperty` and don't make it enumerable?